srtcore/queue.cpp

```diff
@@ -388,14 +364,47 @@ uint64_t CSndUList::getNextProcTime()
    return m_pHeap[0]->m_llTimeStamp_tk;
 }
 
+
+void CSndUList::realloc_()
+{
+   CSNode** temp = NULL;
+
+   try
+   {
+       temp = new CSNode * [m_iArrayLength * 2];
+   }
+   catch (...)
+   {
+       return;
```

Comment:
`CSndBuffer` throws `CUDTException(MJ_SYSTEMRES, MN_MEMORY, 0)`. Probably the same has to be done here.

Reply:
Well, there can be a problem here because this call isn't exactly connected with an API call.

Reply:
Well, I thought it can't be done, but indeed - the only functions from public API of this class is `remove` and `update`, and `update` is the only call that can cause this problem is from `CUDT::sendfile` and `CUDT::sendmsg2`.

Maybe it would be even better idea to move the `insert` function to a private section? If not even remove... When I did so for testing, the code still compiles.

Reply:
`insert` is private now. Renamed to `insert_` and `insert_norealloc_` (according with the naming of the class) and removed the mutex lock.